src/ast_selectors.cpp

```diff
@@ -154,19 +155,17 @@ namespace Sass {
       return false;
     }
 
-    // would like to replace this without stringification
+    // replaced compare without stringification
     // https://github.com/sass/sass/issues/2229
-    // SimpleSelectorSet lset, rset;
-    std::set<std::string> lset, rset;
+    SelectorSet lset, rset;
 
     if (lbase && rbase)
     {
-      if (lbase->to_string() == rbase->to_string()) {
-        for (size_t i = 1, L = length(); i < L; ++i)
-        { lset.insert((*this)[i]->to_string()); }
-        for (size_t i = 1, L = rhs->length(); i < L; ++i)
-        { rset.insert((*rhs)[i]->to_string()); }
-        return includes(rset.begin(), rset.end(), lset.begin(), lset.end());
+      if (*lbase == *rbase) {
+        // create ordered sets for includes query
+        lset.insert(this->begin(), this->end());
+        rset.insert(rhs->begin(), rhs->end());
+        return std::includes(rset.begin(), rset.end(), lset.begin(), lset.end(), OrderSelectors);
```

Comment:
Consider `SelectorSet::key_compare` instead of `OrderSelector` (it might communicate the meaning better here)

Reply:
Can you elaborate a bit more? BTW. I struggled quite a bit here since I tried to reuse `struct OrderNodes` but failed miserably. Not sure if the way one defines the order for a `std::set` is incompatible to pass to `std::includes`, it feels like it should be possible to use the same way somehow??

Reply:
Found it, one can pass `OrderNodes()` directly. Looks like I already found this one a few months ago for `std::sort` but forgot about it. Reverted it in my other PR.